starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,40 @@
 function QueueDataStructure () {
+    this.queueControl =[];
+    this.MAX_SIZE=1;
 }
+
+QueueDataStructure.prototype.isEmpty=function(){
+
+  if (this.queueControl.length===0){
+    return true;
+  } else {
+    return false;
+  }
+};
+QueueDataStructure.prototype.canEnqueue=function(){
+if (this.MAX_SIZE===this.queueControl.length){
+  return false;
+} else {
+  this.dequeue();
+  return true;
+}
+};
+
+QueueDataStructure.prototype.enqueue=function(number){
+  if  (this.canEnqueue()) {
+    this.queueControl.push(number);
```

Comment:
push adds elements to the end, but since this is a queue, you should be adding to the beginning:

_this.queueControl.unshift(number);_